src/util/std_code.h

```diff
@@ -420,6 +420,34 @@ class code_declt:public codet
     return symbol().get_identifier();
   }
 
+  /// Returns the initial value to which the declared variable is initialized,
+  /// or empty in the case where no initialisation is included.
+  optionalt<exprt> initial_value() const
+  {
+    if(operands().size() < 2)
+      return {};
+    return {op1()};
+  }
+
+  /// Sets the value to which this declaration initializes the declared
+  /// variable. Empty optional maybe passed to remove existing initialisation.
+  void set_initial_value(optionalt<exprt> initial_value)
```

Comment:
If these can exist with optional `initial_value`s then I'd much prefer we had a constructor that took an initial value as the argument rather than having a separate setter.

Reply:
All the examples I have found so far were mutating after construction. I could add the facility to set this via the constructor as well.